### ISSUE TITLE
Feature: Dual monitor support

### ISSUE THREAD

**Issue by map7** [kasmtech/KasmVNC#196]
**Is your feature request related to a problem? Please describe.**
Current we use thinlinc for our virtual desktops and it supports multi-monitor. It would be great to see KasmVNC offer such a thing.

**Describe the solution you'd like**
When going to full screen have the option to use all monitors connected to your client machine.

**Comment by mmcclaskey** [comment]
Currently being developed, should be merged into master here in the next few weeks. At that point we will have a developer preview build available.

**Comment by ad56583964** [comment]
、

> Currently being developed, should be merged into master here in the next few weeks. At that point we will have a developer preview build available.

That's awesome to hear! On a related note, I was wondering if there could be a feature to create an 'unlimited number' of virtual screens within a single-user scenario, even with just one or two physical monitors. This might add a whole new level of flexibility to our workspace. Could be a neat addition alongside the multi-monitor support, especially if it can be integrated well with XFCE. Just throwing this idea out there!"

**Comment by dstromberg** [comment]
I'm eager to try this.

The subject says "dual" - will it also work with 3 displays of different sizes?

**Comment by mmcclaskey** [comment]
It will support N number of displays. Here is a preview build, we are currently trying to get this merged into the master branch. That being said, please keep in mind that it is pre-beta. Please provide any feedback. You will see a new primary button in the control panel that will open up a display manager. New screens added will open up a new tab in your browser, move that tab into a new window and place where you want in your physical setup. Then adjust the location in the KasmVNC display manager so that KasmVNC knows where the windows are in relation to each other. Enjoy...

[Bionic amd64](https://kasmweb-build-artifacts.s3.amazonaws.com/kasmvnc/f712f5e7f8e58e3a77354e43cc68027d03d08a98/kasmvncserver_bionic_1.2.1_feature_KASM-5078_multi_monitor_f712f5_amd64.deb)
[Bionic arm64](https://kasmweb-build-artifacts.s3.amazonaws.com/kasmvnc/f712f5e7f8e58e3a77354e43cc68027d03d08a98/kasmvncserver_bionic_1.2.1_feature_KASM-5078_multi_monitor_f712f5_arm64.deb)
[Bookworm amd64](https://kasmweb-build-artifacts.s3.amazonaws.com/kasmvnc/f712f5e7f8e58e3a77354e43cc68027d03d08a98/kasmvncserver_bookworm_1.2.1_feature_KASM-5078_multi_monitor_f712f5_amd64.deb)
[Bookworm arm64](https://kasmweb-build-artifacts.s3.amazonaws.com/kasmvnc/f712f5e7f8e58e3a77354e43cc68027d03d08a98/kasmvncserver_bookworm_1.2.1_feature_KASM-5078_multi_monitor_f712f5_arm64.deb)
[Bullseye amd64](https://kasmweb-build-artifacts.s3.amazonaws.com/kasmvnc/f712f5e7f8e58e3a77354e43cc68027d03d08a98/kasmvncserver_bullseye_1.2.1_feature_KASM-5078_multi_monitor_f712f5_amd64.deb)
[Bullseye arm64](https://kasmweb-build-artifacts.s3.amazonaws.com/kasmvnc/f712f5e7f8e58e3a77354e43cc68027d03d08a98/kasmvncserver_bullseye_1.2.1_feature_KASM-5078_multi_monitor_f712f5_arm64.deb)
[Buster amd64](https://kasmweb-build-artifacts.s3.amazonaws.com/kasmvnc/f712f5e7f8e58e3a77354e43cc68027d03d08a98/kasmvncserver_buster_1.2.1_feature_KASM-5078_multi_monitor_f712f5_amd64.deb)
[Buster arm64](https://kasmweb-build-artifacts.s3.amazonaws.com/kasmvnc/f712f5e7f8e58e3a77354e43cc68027d03d08a98/kasmvncserver_buster_1.2.1_feature_KASM-5078_multi_monitor_f712f5_arm64.deb)
[CentOS Core amd64](https://kasmweb-build-artifacts.s3.amazonaws.com/kasmvnc/f712f5e7f8e58e3a77354e43cc68027d03d08a98/kasmvncserver_centos_core_1.2.1_feature_KASM-5078_multi_monitor_f712f5_x86_64.rpm)
[Fedora 38 aarch64](https://kasmweb-build-artifacts.s3.amazonaws.com/kasmvnc/f712f5e7f8e58e3a77354e43cc68027d03d08a98/kasmvncserver_fedora_thirtyeight_1.2.1_feature_KASM-5078_multi_monitor_f712f5_aarch64.rpm)
[Fedora 38 amd64](https://kasmweb-build-artifacts.s3.amazonaws.com/kasmvnc/f712f5e7f8e58e3a77354e43cc68027d03d08a98/kasmvncserver_fedora_thirtyeight_1.2.1_feature_KASM-5078_multi_monitor_f712f5_x86_64.rpm)
[Fedora 37 aarch64](https://kasmweb-build-artifacts.s3.amazonaws.com/kasmvnc/f712f5e7f8e58e3a77354e43cc68027d03d08a98/kasmvncserver_fedora_thirtyseven_1.2.1_feature_KASM-5078_multi_monitor_f712f5_aarch64.rpm)
[Fedora 37 amd64](https://kasmweb-build-artifacts.s3.amazonaws.com/kasmvnc/f712f5e7f8e58e3a77354e43cc68027d03d08a98/kasmvncserver_fedora_thirtyseven_1.2.1_feature_KASM-5078_multi_monitor_f712f5_x86_64.rpm)
[Focal amd64](https://kasmweb-build-artifacts.s3.amazonaws.com/kasmvnc/f712f5e7f8e58e3a77354e43cc68027d03d08a98/kasmvncserver_focal_1.2.1_feature_KASM-5078_multi_monitor_f712f5_amd64.deb)
[Focal arm64](https://kasmweb-build-artifacts.s3.amazonaws.com/kasmvnc/f712f5e7f8e58e3a77354e43cc68027d03d08a98/kasmvncserver_focal_1.2.1_feature_KASM-5078_multi_monitor_f712f5_arm64.deb)
[Jammy amd64](https://kasmweb-build-artifacts.s3.amazonaws.com/kasmvnc/f712f5e7f8e58e3a77354e43cc68027d03d08a98/kasmvncserver_jammy_1.2.1_feature_KASM-5078_multi_monitor_f712f5_amd64.deb)
[Jammy arm64](https://kasmweb-build-artifacts.s3.amazonaws.com/kasmvnc/f712f5e7f8e58e3a77354e43cc68027d03d08a98/kasmvncserver_jammy_1.2.1_feature_KASM-5078_multi_monitor_f712f5_arm64.deb)
[Kali amd64](https://kasmweb-build-artifacts.s3.amazonaws.com/kasmvnc/f712f5e7f8e58e3a77354e43cc68027d03d08a98/kasmvncserver_kali-rolling_1.2.1_feature_KASM-5078_multi_monitor_f712f5_amd64.deb)
[Kali arm64](https://kasmweb-build-artifacts.s3.amazonaws.com/kasmvnc/f712f5e7f8e58e3a77354e43cc68027d03d08a98/kasmvncserver_kali-rolling_1.2.1_feature_KASM-5078_multi_monitor_f712f5_arm64.deb)
[Oracle 8 amd64](https://kasmweb-build-artifacts.s3.amazonaws.com/kasmvnc/f712f5e7f8e58e3a77354e43cc68027d03d08a98/kasmvncserver_oracle_8_1.2.1_feature_KASM-5078_multi_monitor_f712f5_aarch64.rpm)
[Oracle 8 arm64](https://kasmweb-build-artifacts.s3.amazonaws.com/kasmvnc/f712f5e7f8e58e3a77354e43cc68027d03d08a98/kasmvncserver_oracle_8_1.2.1_feature_KASM-5078_multi_monitor_f712f5_x86_64.rpm)

**Comment by dstromberg** [comment]
I tried to install the preview build, but got:
```
$ sudo apt-get install ./kasmvncserver_bookworm_1.2.1_feature_KASM-5078_multi_monitor_f712f5_amd64.deb libwebp7
below cmd output started 2023 Tue Dec 19 01:15:42 PM PST
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Note, selecting 'kasmvncserver' instead of './kasmvncserver_bookworm_1.2.1_feature_KASM-5078_multi_monitor_f712f5_amd64.deb'
libwebp7 is already the newest version (1.2.2-2ubuntu0.22.04.2).
kasmvncserver is already the newest version (1.2.0-1).
You might want to run 'apt --fix-broken install' to correct these.
The following packages have unmet dependencies:
 kasmvncserver : Depends: libwebp7 (>= 1.2.4) but 1.2.2-2ubuntu0.22.04.2 is to be installed
E: Unmet dependencies. Try 'apt --fix-broken install' with no packages (or specify a solution).
```

I'm on an Ubuntu system:
```
$ lsb_release -a
below cmd output started 2023 Tue Dec 19 01:16:50 PM PST
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 22.04.2 LTS
Release:        22.04
Codename:       jammy
above cmd output done    2023 Tue Dec 19 01:16:50 PM PST
dstromberg@usbopl-vdi001:/tmp x86_64-pc-linux-gnu 3381685

$ cat /etc/debian_version
below cmd output started 2023 Tue Dec 19 01:16:54 PM PST
bookworm/sid

```

Would it be easier to build from source?

**Comment by mmcclaskey** [comment]
You look to have installed the Debian Bookworm package. Download and install the Ubuntu Jammy one.

**Comment by dstromberg** [comment]
Installing the Ubuntu Jammy package helped a lot.

I'm on a MacOS 14.2.1 (upgraded it this morning) system using Chrome Version 120.0.6099.129 (Official Build) (x86_64) as my browser.  I have "Displays have separate spaces" selected in Apple -> Settings -> Desktop & Dock; without that one display would fullscreen, and the other 2 would go solid black.

I can get 3 displays now - that's great.

Sadly, when I fullscreen in the KasmVNC menu, only one of the 3 displays goes fullscreen.  The other 2 still show the traffic light buttons, tabs, omnibar and bookmarks at the top, and MacOS' dock is still there at the bottom on one of them.

Is there a fix for this?

Thanks!

**Comment by dstromberg** [comment]
I'm thinking maybe I could fullscreen each browser window individually, but then how would I un-fullscreen?  One would have a menu for it, but the other two?

**Comment by mmcclaskey** [comment]
Until we get a better solution in place, you can work around this by using the full screen shortcut, on MacOS that would be **Control + Command + F**

**Comment by dstromberg** [comment]
Sadly, Control+Command+F doesn't appear to be working on my system.  It acts as though I didn't type anything.  I tried both going-to-fullscreen, and leaving-fullscreen - neither did anything.

I'm on a Mac laptop with a PC keyboard attached; I tried both the built-in keyboard (with control and command keys), and the PC keyboard (I substituted the Windows key for command).

Both keyboards are working for most keystrokes.

Thanks for your hard work on this!
